src/yki/registration/registration.clj

```diff
@@ -77,33 +89,41 @@
     (log/info "END: Init exam session" exam_session_id "registration success" registration-id)
     (ok response)))
 
+;; Check if person is eligible to register
+(defn- is-person-eligible [db participant-id exam_session_id ssn spacef]
+  (let [space-left?        (spacef db exam_session_id nil)
+        not-registered?    (registration-db/not-registered-to-exam-session? db participant-id exam_session_id)
+        ssn-doenst-exists? (registration-db/previous-ssn-doesnt-exists? db ssn exam_session_id)]
+    (log/info "validity check" space-left? not-registered? ssn-doenst-exists?)
+  {:space-left space-left?
+    :not-registered not-registered?
+    :ssn-doesnt-exists ssn-doenst-exists?}
+  ))
+;; Provides login flow
 (defn init-registration
   [db session {:keys [exam_session_id]} payment-config]
   (log/info "START: Init exam session" exam_session_id "registration")
-  (let [participant-id (get-or-create-participant db (:identity session))
+  (let [reg-identity (:identity session)
+        participant-id (get-or-create-participant db reg-identity)
         started-registration-id (registration-db/started-registration-id-by-participant db participant-id exam_session_id)]
     (log/warn "started-registration-id" started-registration-id)
     (if started-registration-id
       (ok (create-init-response db session exam_session_id started-registration-id payment-config))
       (cond
         ; admission open
         (registration-db/exam-session-registration-open? db exam_session_id)
-        (let [space-left?        (registration-db/exam-session-space-left? db exam_session_id nil)
-              not-registered?    (registration-db/not-registered-to-exam-session? db participant-id exam_session_id)]
-          (if (and space-left? not-registered?)
+        (let [eligible        (is-person-eligible db participant-id exam_session_id (:ssn reg-identity) registration-db/exam-session-space-left?)]
+          (if (every?  identity ( vals eligible))
```

Comment:
Käytetäänkö is-person-eligible vastausta aina näin? Jos käytetään, selkeintä jos se palauttaisi samoin tein booleanin.

Reply:
Yksittäisiä tietoja käytetään error vastauksessa. jotta ui tietää vastata oikein.

Reply:
Niinpäs onkin 👍